src/Interceptor/RetryRequests.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Amp\Http\Client\Interceptor;
+
+use Amp\CancellationToken;
+use Amp\Http\Client\ApplicationInterceptor;
+use Amp\Http\Client\Client;
+use Amp\Http\Client\Connection\UnprocessedRequestException;
+use Amp\Http\Client\HttpException;
+use Amp\Http\Client\Request;
+use Amp\Promise;
+use function Amp\call;
+
+final class RetryRequests implements ApplicationInterceptor
+{
+    /** @var int */
+    private $retryLimit;
+
+    public function __construct(int $retryLimit = 2)
```

Comment:
The limit here should probably match the one in `Client`, maybe skip providing a default value here.

Reply:
Oops, not matching was a typo. Required in the interceptor makes sense.